### DISCUSSION TITLE
ElementsController::_asFailure calls asModelFailure

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/cms#11807] [Ideas] [answered]
@brandonkelly 

The `ElementsController::_asSuccess()` method includes a comment:

> // Don't call asModelSuccess() here so we can avoid including custom fields in the element data

See: https://github.com/craftcms/cms/blob/develop/src/controllers/ElementsController.php#L1691

But the `ElementsController::_asFailure()` method calls `Controller::asModelFailure()`: https://github.com/craftcms/cms/blob/develop/src/controllers/ElementsController.php#L1736

Both `Controller::asModelSuccess()` and `Controller::asModelFailure()` call `$model->toArray()` which is what you're trying to avoid in `ElementsController::_asSuccess`.

Shouldn't it reason that `ElementsController::_asFailure()` should not call `Controller::asModelFailure()` and instead construct the model return data without custom fields:

```
$element->toArray($element->attributes())
```

The only thing I can think of is that returning the whole model might return errors?

**Comment by brandonkelly** [comment, marked as answer]
Yeah, that should have been updated as well, thanks for pointing that out! Done for the next release (dc11cf32ae813aacfe1fefc98f325668a277b600).

**Reply by brandonkelly** [reply]
4.2.2 is out with that change.
